gen/calc.py

```diff
@@ -547,6 +547,53 @@ def validate_exhibitor_storage_master_discovery(master_discovery, exhibitor_stor
             "`master_http_load_balancer` then exhibitor_storage_backend must not be static."
 
 
+def calculate_adminrouter_tls_version_override(
+        adminrouter_tls_1_0_enabled,
+        adminrouter_tls_1_1_enabled,
+        adminrouter_tls_1_2_enabled):
+    tls_versions = list()
+    if adminrouter_tls_1_0_enabled == 'true':
+        tls_versions.append('TLSv1')
+
+    if adminrouter_tls_1_1_enabled == 'true':
+        tls_versions.append('TLSv1.1')
+
+    if adminrouter_tls_1_2_enabled == 'true':
+        tls_versions.append('TLSv1.2')
+
+    tls_version_string = " ".join(tls_versions)
+    return tls_version_string
+
+
+def calculate_adminrouter_tls_cipher_override(adminrouter_tls_cipher_suite):
+    if adminrouter_tls_cipher_suite != '':
+        return 'true'
+    else:
+        return 'false'
+
+
+def validate_adminrouter_tls_version_present(
+        adminrouter_tls_1_0_enabled,
+        adminrouter_tls_1_1_enabled,
+        adminrouter_tls_1_2_enabled):
+
+    tls_version_flags = [
+        adminrouter_tls_1_0_enabled,
+        adminrouter_tls_1_1_enabled,
+        adminrouter_tls_1_2_enabled,
+    ]
+
+    enabled_tls_flags_count = len(
+        [flag for flag in tls_version_flags if flag == 'true'])
+
+    msg = (
+        'At least one tls boolean (adminrouter_tls_1_0_enabled, '
+        'adminrouter_tls_1_1_enabled, adminrouter_tls_1_2_enabled) must '
+        'be set.'
```

Comment:
```s/be set/be set to `'true` `` or similar - you can set them to "false".

Reply:
While we are here:I prefer "At least one of" instead of "At least one tls boolean".

Reply:
(can be done in a follow-up!!)